lambda/matching.ml

```diff
@@ -2752,57 +2752,42 @@ let mk_failaction_neg partial ctx def =
     )
   | Total -> (None, Jumps.empty)
 
-(* In line with the article and simpler than before *)
+(* In [mk_failaction_pos partial seen ctx defs],
+   - [partial] is Total if we know the current switch
+     is already exhaustive
+   - [seen] is the list of constructors accepted by the switch
+     (those that will be matched)
+   - [ctx] is the current context (what we know of the value
+     being matched)
+   - [defs] is the default environment (what inputs
+     are expected by the switches present at larger exit numbers).
+
+   The function returns a triple [(fail, fails, jumps)] containing
+   information for the failure cases, the constructors missing from
+   the current switch:
+   - [fail] is an optional 'default' action for the switch
+   - [fails] is a list of extra switch clauses to add for failure cases
```

Comment:
Perhaps a missing detail is that the action for `fails` is always a raise to a larger exit number. (And that the "accuracy" mentioned below arises from skipping past unnecessary exit handlers.)